ext/hash/hash.stub.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/** @return string|false */
+function hash(string $algo, string $data, bool $raw_output = false) {}
+
+/** @return string|false */
+function hash_file(string $algo, string $filename, bool $raw_output = false) {}
+
+/** @return string|false */
+function hash_hmac(string $algo, string $data, string $key, bool $raw_output = false) {}
+
+/** @return string|false */
+function hash_hmac_file(string $algo, string $data, string $key, bool $raw_output = false) {}
+
+function hash_init(string $algo, int $options = 0, ?string $key = null): HashContext {}
+
+/** @return true */
+function hash_update(HashContext $context, string $data): bool {}
+
+/** @param resource $handle */
+function hash_update_stream(HashContext $context, $handle, int $length = -1): int {}
+
+/** @param resource $stream_context */
+function hash_update_file(HashContext $context, string $filename, $stream_context): bool {}
+
+function hash_final(HashContext $context, bool $raw_output = false): string {}
+
+function hash_copy(HashContext $context): HashContext {}
+
+function hash_algos(): array {}
+
+function hash_pbkdf2(string $algo, string $password, string $salt, int $iterations, int $length = 0, bool $raw_output = false): string {}
+
+function hash_equals(string $known_string, string $user_string) : bool {}
```

Comment:
Hmm, these `string`s have to really be strings (no type juggling is performed). Not sure if that should be explicitly mentioned in the doc block.

Reply:
Yup, I was not really sure what was the right decision here.
For now, I have dropped the explicit types from the function signature (it matches the existing arginfo structure) and added a doc block.

Reply:
Weird idea but what about specifying ``declare("strict-types"=1)`` to convey this information?

Reply:
`declare(strict-types=1)` might be confusing, since it normally affects the calling code. Anyhow, I think keeping the `string` type declarations is sensible (after all, strings are accepted, but not arrays, for example), and additionally having a brief note in the doc block. I do this minor correction myself, and going to merge. Thanks.